### DISCUSSION TITLE
Consider saving add.xml in memory and save to disk only when actually saving

### DISCUSSION THREAD

**Discussion by Luffyyy** [diesel-modding/BeardLib-Editor#681] [Ideas]
The way add.xml currently works is reminiscent of the old way we used to load assets which was by exporting from extract this has *mostly* disappeared in favor of loading through the database, removing the need to include any file in the map. Including files in the map sorta required us to also make sure the add.xml is updated correctly. However, this is no longer needed.

We should account for 2 cases: 

## Not including the files from database
In this situation it should be *mostly* safe to keep in memory without doing much extra before saving the map.

## Including the files from database
In this situation we should consider moving the files only on save and not before to avoid trash files. There are cases we need to include files however compared to before we *can* load the assets beforehand much easily (though previously it was possible, just not considered)

That being said, I'm not sure if there are some cases in which a map isn't exactly saved but if we save something into memory and not save then the map might break after loading it again.

This however doesn't deal with a situation where you'd delete a unit or an asset after saving however it should reduce the amount of unused assets by saving the add.xml less.

**Comment by Dushess** [comment]
I think we need this. If you searching many units to place in some cases, the add.xml will be clogged very fast. 
We need to be sure there's no scenario left, when you starting the level before saving required stuff. And if keeping too much work in memory - it might be suddenly lost, and we can't have option to save it quickly first, actually lost power later. An example... 

I don't really got 1st way, Add from DB will be used but not autosaved or what?
